contracts/Strategy.sol

```diff
@@ -16,32 +16,119 @@ import {
+        address _rewards,
+        address _keeper,
+        address _bentoBox,
+        address _kashiPair
+    ) public {
+        _initialize(_vault, _strategist, _rewards, _keeper);
+        _initializeStrat(_bentoBox, _kashiPair);
+    }
+
+    event Cloned(address indexed clone);
+
+    function cloneKashiLender(
+        address _vault,
+        address _strategist,
+        address _rewards,
+        address _keeper,
+        address _bentoBox,
+        address _kashiPair
+    ) external returns (address newStrategy) {
+        require(isOriginal);
+        // Copied from https://github.com/optionality/clone-factory/blob/master/contracts/CloneFactory.sol
+        bytes20 addressBytes = bytes20(address(this));
+        assembly {
+            // EIP-1167 bytecode
+            let clone_code := mload(0x40)
+            mstore(
+                clone_code,
+                0x3d602d80600a3d3981f3363d3d373d3d3d363d73000000000000000000000000
+            )
+            mstore(add(clone_code, 0x14), addressBytes)
+            mstore(
+                add(clone_code, 0x28),
+                0x5af43d82803e903d91602b57fd5bf30000000000000000000000000000000000
+            )
+            newStrategy := create(0, clone_code, 0x37)
+        }
+
+        Strategy(newStrategy).initialize(
+            _vault,
+            _strategist,
+            _rewards,
+            _keeper,
+            _bentoBox,
+            _kashiPair
+        );
+
+        emit Cloned(newStrategy);
+    }
+
+    function _initializeStrat(address _bentoBox, address _kashiPair) internal {
+        require(
+            address(kashiPair) == address(0),
+            "StategyKashiLending: already initialized"
+        );
+        require(address(IKashiPair(_kashiPair).bentoBox()) == _bentoBox);
+        require(address(IKashiPair(_kashiPair).asset()) == address(want));
+
+        bentoBox = IBentoBox(_bentoBox);
+        kashiPair = IKashiPair(_kashiPair);
+
+        want.safeApprove(_bentoBox, type(uint256).max);
+    }
 
     function name() external view override returns (string memory) {
-        // Add your own name here, suggestion e.g. "StrategyCreamYFI"
-        return "Strategy<ProtocolName><TokenType>";
+        return "StrategyKashiLender";
     }
 
     function estimatedTotalAssets() public view override returns (uint256) {
-        // TODO: Build a more accurate estimate using the value of all positions in terms of `want`
-        return want.balanceOf(address(this));
+        uint256 sharesInBento =
+            bentoBox.balanceOf(BIERC20(address(want)), address(this));
+        uint256 kashiTokens = kashiPair.balanceOf(address(this));
+
+        uint256 totalShares =
+            sharesInBento.add(_toBentoShares(kashiTokens, true));
+
+        return
+            want.balanceOf(address(this)).add(
+                bentoBox.toAmount(BIERC20(address(want)), totalShares, true)
+            );
     }
 
     function prepareReturn(uint256 _debtOutstanding)
```

Comment:
Why does the strat not support farming and selling sushi rewards for the kashi lp tokens for more want?

Reply:
No farming rewards for kashi yet

Reply:
There is farming rewards though,check the yield tab and type usdc or 1inch,you can see wkmxys/xyz lp for farming sushi

Reply:
Those are for LPs not kashi

Reply:
![image](https://user-images.githubusercontent.com/25401217/120680437-f8a15200-c4a2-11eb-97ad-d7526e1875b0.png)


See those kmxyz yield pairs?those are kashi lp tokens one can stake

Reply:
Oh, very nice. I'll investigate adding that ASAP.

Reply:
For the initial testing on Polygon/BSC and for the RAI vault, there are no masterchef rewards, so I'm holding off on this for now. Will add in a future iteration.